test/framework/alertmanager.go

```diff
@@ -221,25 +222,31 @@ func (f *Framework) WaitForAlertmanagerInitialized(ns, name string, amountPeers
 		}
 
 		isAlertmanagerInClusterMode := amountPeers > 1 || forceEnableClusterMode
-		if !isAlertmanagerInClusterMode && amStatus.Status == "success" {
+		if !isAlertmanagerInClusterMode {
 			return true, nil
 		}
 
-		if amStatus.Data.ClusterStatus == nil {
+		if amStatus.Cluster == nil {
 			pollError = fmt.Errorf("do not have a cluster status")
 			return false, nil
 		}
 
-		if len(amStatus.Data.ClusterStatus.Peers) == amountPeers {
-			return true, nil
+		if *amStatus.Cluster.Status != "ready" {
+			pollError = fmt.Errorf("failed to get cluser status, expected ready, got %s", *amStatus.Cluster.Status)
+			return false, nil
+		}
+
+		if len(amStatus.Cluster.Peers) != amountPeers {
+			pollError = fmt.Errorf("failed to get correct amount of peers, expected %d, got %d", amountPeers, len(amStatus.Cluster.Peers))
+			return false, nil
 
 		}
 
 		pollError = fmt.Errorf(
```

Comment:
you need to remove this line and have `return true, nil` as the last statement because when we reach this point, all checks have passed.